fulltext/extract.py

```diff
@@ -0,0 +1,66 @@
+"""Provides asynchronous task for fulltext extraction."""
+
+from celery import shared_task
+import os
+from datetime import datetime
+from fulltext import logging
+from fulltext.services.fulltext import extractor
+from fulltext.services.store import store
+from fulltext.services.metrics import metrics
+from fulltext.services.retrieve import retrievePDF
+from celery.result import AsyncResult
+from celery import current_app
+from celery.signals import after_task_publish
+logger = logging.getLogger(__name__)
+
+
+@shared_task
+def extract_fulltext(document_id: str, pdf_url: str) -> None:
+    """Perform fulltext extraction for a single arXiv document."""
+    logger.info('Retrieving PDF for %s' % document_id)
+    start_time = datetime.now()
+    try:
+        # Retrieve PDF from arXiv central document store.
+        pdf_path = retrievePDF.session.retrieve(pdf_url, document_id)
+        if pdf_path is None:
+            metrics.session.report('PDFIsAvailable', 0.)
+            msg = '%s: no PDF available' % document_id
+            logger.info(msg)
+            raise RuntimeError(msg)
```

Comment:
Would a FileNotFoundError or custom exception be more informative if we like this exception control flow pattern?